x/mongo/driver/compression.go

```diff
@@ -105,8 +127,15 @@ func CompressPayload(in []byte, opts CompressionOpts) ([]byte, error) {
 	}
 }
 
+var zstdReaderPool = sync.Pool{
+	New: func() interface{} {
+		r, _ := zstd.NewReader(nil)
+		return r
```

Comment:
While we don't expect this to ever return an error with the current implementation, this could cause a difficult-to-diagnose bug if anyone ever changed the `NewReader` params. It's safer to check the error and panic.

```suggestion
		r, err := zstd.NewReader(nil)
		if err != nil {
			panic(err)
		}
		return r
```

Reply:
I don't think we need this since our tests cover this case and will panic if this ever returns `nil` so any future changes to `zstd.NewReader` will be caught by the tests when we upgrade (I'll add a regression test just in case). 

I'm also pretty adamant that libraries should never panic.

If for some reason we really wanted to check this error then we could use the below:
```go
func getZstdReader() (*zstd.Decoder, error) {
	if r, _ := zstdReaderPool.Get().(*zstd.Decoder); r != nil {
		return r, nil
	}
	return zstd.NewReader(nil) 
}
```

Reply:
If this is covered by a test case, then that sounds reasonable. Consider this resolved.